frontend/src/pages/post/[slug].tsx

```diff
@@ -1,87 +1,162 @@
 
 interface Params extends ParsedUrlQuery {
-    slug: string;
+  slug: string;
 }
 
-interface Props {
-    post: PostDetail;
-    fallbackBanner: string;
+export interface PostPageProps {
+  post: PostDetail;
+  fallbackBanner: string;
 }
 
-export default function PostPage(props: Props) {
-    const post = props.post;
-    
-    return <div>
-        <HeadArticle post={post}/>
-        <Banner title={post.title}
-                subtitle={<>
-                    {post.author}
-                    <span className="mx-2">&#xB7;</span>
-                    {dateToText(new Date(post.publishedAt))}
-                </>}
-                background={post.banner ?? props.fallbackBanner}
-                compact/>
-        <Main className="container">
-            <p className="text-faded mb-4">
-                <PostTypeDisplay type={props.post.type}/>
-                <span className="mx-2">&#xB7;</span>
-                <span>
-                    <Link href={`/auteur/${encodeURIComponent(post.author)}`}>
-                        <a className="hover:underline hover:text-primary transition-colors">
-                            {post.author}
-                        </a>
-                    </Link>
-                </span>
-                <span className="mx-2">&#xB7;</span>
-                <span>{dateToText(new Date(post.publishedAt))}</span>
-                {post.type === PostType.SUBMISSION &&
-                    <p className="mt-4">
-                        <i>
-                            Dit artikel is een ingezonden opiniestuk. De opvattingen zijn daarmee geen officiële standpunten van ROOD en sluiten niet
-                            noodzakelijkerwijs aan op de meningen van al onze leden.
-                        </i>
-                    </p>}
+interface PetitionProps {
+  post: PostDetail;
+  petition: PetitionDetail;
+  fallbackBanner: string;
+}
+
+function PetitionBody(props: PetitionProps) {
+  const petition = props.petition;
+  const post = props.post;
+
+  const { asPath } = useRouter();
+  const [signThanks, setSignThanks] = useState(<></>);
+
+  useEffect(() => {
+    const hash = asPath.split("#")[1];
+    if (hash && hash.includes("thanks")) {
+      setSignThanks(
+        <div className="mt-3 font-title text-2xl font-bold bg-primary text-white container">
+          Bedankt voor het ondertekenen! Misschien kun je de petitie nu delen?
+        </div>
+      );
+    }
+  }, [asPath]);
+
+  return (
+    <>
+      {signThanks}
+      <Main className="pt-8 container md:max-w-screen-xl mx-auto flex flex-col md:flex-row gap-4">
+        <div className="md:basis-4/12 md:ml-auto">
+          <PetitionCard
+            petition={petition}
+            className="md:sticky md:top-[90px]"
+          />
+        </div>
+        <div className="md:basis-8/12 md:order-first">
+          <Markdown content={post.content} />
+        </div>
+      </Main>
+    </>
+  );
+}
+
+function PostBody(props: PostPageProps) {
```

Comment:
Dit moet ook in een apart bestand, maar zou het mooier vinden als je het samen kon houden omdat er weinig verschil is tussen een petitie en een ander soort post naast het petition-blok dat verschijnt. Dat kan je echter ook met een conditional render afvangen i.p.v. veel potentiële duplicate code te introduceren